lint top-level JS files

Currently the configuration files do not follow the same lint rules as the rest of the source:

```
$ eslint src *.js

angularjs-webpack-seed/karma.conf.js
   7:1  warning  Unexpected 'todo' comment           no-warning-comments
   7:1  warning  Unexpected ' TODO' comment          no-warning-comments
  10:1  error    More than 1 blank line not allowed  no-multiple-empty-lines
  44:7  error    Extra space after key 'type'        key-spacing
  45:7  error    Extra space after key 'dir'         key-spacing

angularjs-webpack-seed/webpack.config.common.js
  24:10  error  Unexpected trailing comma                comma-dangle
  33:14  error  Extra space before value for key 'test'  key-spacing

angularjs-webpack-seed/webpack.config.develop.js
   6:11  error  Use path.join() or path.resolve() instead of + to create paths  no-path-concat
  11:25  error  Unexpected trailing comma                                       comma-dangle
  24:11  error  Missing space before value for key 'proxy'                      key-spacing

angularjs-webpack-seed/webpack.config.prod.js
   2:35  error  Strings must use singlequote                                    quotes
  13:11  error  Use path.join() or path.resolve() instead of + to create paths  no-path-concat
  16:47  error  Unexpected trailing comma                                       comma-dangle
  26:17  error  Extra space after key 'screw_ie8'                               key-spacing
  26:17  error  Identifier 'screw_ie8' is not in camel case                     camelcase
  26:35  error  Identifier 'keep_fnames' is not in camel case                   camelcase
  27:19  error  Identifier 'screw_ie8' is not in camel case                     camelcase

✖ 17 problems (15 errors, 2 warnings)
```

--- a/package.json
+++ b/package.json
@@ -4,7 +4,7 @@
   "description": "A starter kit for using AngularJS with Webpack and ES6",
   "main": "src/index.js",
   "scripts": {
-    "build": "rimraf ./build && webpack --config ./webpack.config.prod.js",
+    "build": "rimraf ./build && eslint *.js && webpack --config ./webpack.config.prod.js",
     "develop": "webpack-dev-server --config ./webpack.config.develop.js --open",
     "serve": "webpack --config ./webpack.config.prod.js -p --progress --profile --bail && http-server ./build -o",
     "test:unit": "rimraf ./reports && karma start"

--- a/webpack.config.common.js
+++ b/webpack.config.common.js
@@ -21,7 +21,7 @@ module.exports = {
           presets: [
             ['es2015', { modules: false }]
           ]
-        },
+        }
       }, {
         loader: 'eslint-loader',
         options: {
@@ -30,7 +30,7 @@ module.exports = {
         }
       }]
     }, {
-      test:  /\.html$/,
+      test: /\.html$/,
       use: [{
         loader: 'html-loader'
       }]

--- a/webpack.config.develop.js
+++ b/webpack.config.develop.js
@@ -1,14 +1,15 @@
+const path = require('path');
 const webpackMerge = require('webpack-merge');
 const commonConfig = require('./webpack.config.common.js');
 
 module.exports = webpackMerge(commonConfig, {
   output: {
-    path: __dirname + './build',
+    path: path.join(__dirname, 'build'),
     filename: '[name].bundle.js',
     sourceMapFilename: '[name].map',
     chunkFilename: '[id].chunk.js',
     library: 'ac_[name]',
-    libraryTarget: 'var',
+    libraryTarget: 'var'
   },
 
   devServer: {
@@ -21,7 +22,7 @@ module.exports = webpackMerge(commonConfig, {
       aggregateTimeout: 300,
       poll: 1000
     },
-    proxy:{
+    proxy: {
       '/api/*': {
         target: 'http://<your-domain>',
         secure: false,

--- a/webpack.config.prod.js
+++ b/webpack.config.prod.js
@@ -1,5 +1,6 @@
+const path = require('path');
 const commonConfig = require('./webpack.config.common');
-const ExtractTextPlugin = require("extract-text-webpack-plugin");
+const ExtractTextPlugin = require('extract-text-webpack-plugin');
 const UglifyJsPlugin = require('webpack/lib/optimize/UglifyJsPlugin');
 const WebpackMd5Hash = require('webpack-md5-hash');
 const webpackMerge = require('webpack-merge');
@@ -10,10 +11,10 @@ module.exports = webpackMerge(commonConfig, {
   devtool: 'source-map',
 
   output: {
-    path: __dirname + '/build',
+    path: path.join(__dirname, 'build'),
     filename: '[name].[chunkhash].bundle.js',
     sourceMapFilename: '[name].[chunkhash].bundle.map',
-    chunkFilename: '[id].[chunkhash].chunk.js',
+    chunkFilename: '[id].[chunkhash].chunk.js'
   },
 
   plugins: [
@@ -23,8 +24,8 @@ module.exports = webpackMerge(commonConfig, {
     new WebpackMd5Hash(),
     new UglifyJsPlugin({
       beautify: false,
-      mangle: { screw_ie8 : true, keep_fnames: true },
-      compress: { screw_ie8: true },
+      mangle: { screw_ie8: true, keep_fnames: true }, // eslint-disable-line camelcase
+      compress: { screw_ie8: true }, // eslint-disable-line camelcase
       comments: false,
       sourceMap: true
     }),

